### PR TITLE
feat: add string interpolation support for locale service

### PR DESCRIPTION
**Pull request by jikkai**
- [x] I am sure that the code is update-to-date with the `dev` branch.

## Usage

```typescript
const locales = {
  welcome: 'hello, {0}'
}

console.log(t('welcome', 'world')) // 'hello, world'
```

### CODE REVIEW

File: packages/core/src/services/locale/locale.service.ts [hunk 1]
```diff
@@ -63,29 +41,53 @@ export class LocaleService extends Disposable {
...
+                    return resolveKeyPath(nextObj as ILanguagePack, keys);
+                } else {
+                    return nextObj;
+                }
+            }
+
+            return null;
+        }
+
+        // 使用点分隔符拆分key
```
Comment (wzhudev) [inline packages/core/src/services/locale/locale.service.ts:69]: Use English to comment.

File: packages/core/src/services/locale/__tests__/locale.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+describe('LocaleService', () => {
+    let localeService: LocaleService;
+    const testLocales = {
+        [LocaleType.ZH_CN]: {
+            greeting: '你好',
+            farewell: '再见',
+            nested: {
+                welcome: '欢迎光临',
+            },
+            withParams: '你好, {0}!',
```
Comment (wzhudev) [inline packages/core/src/services/locale/__tests__/locale.service.spec.ts:30]: Should add some documentation somewhere to introduce this kind of syntax.

**Review by wzhudev — APPROVED** [review]
Please use English to comment our code.

Overall, LGTM.

**Comment by jikkai** [comment]
I added JSDoc to explain the new syntax of the `t` function.

**Pull request merged**